src/maidsafe/routing/connection_manager.cc

```diff
@@ -146,42 +149,42 @@ void ConnectionManager::AddNode(optional<NodeInfo> assumend_node_info, EndpointP
   auto socket = pair_i->second;
   weak_ptr<crux::socket> weak_socket = socket;
 
-  socket->async_connect
-      (convert::ToBoost(eps.external),
-       [=](boost::system::error_code error) {
-         auto socket = weak_socket.lock();
-
-         if (!socket) return;
+  socket->async_connect(convert::ToBoost(eps.external), [=](boost::system::error_code error) {
+    auto socket = weak_socket.lock();
 
-         if (error) {
-           being_connected_.erase(endpoint);
-           return;
-         }
+    if (!socket)
+      return;
 
-         NodeInfo our_node_info(our_id_, our_fob_);
+    if (error) {
+      being_connected_.erase(endpoint);
+      return;
+    }
 
-         AsyncExchange(*socket, Serialise(our_node_info),
-           [=](boost::system::error_code error, Bytes data) {
-             auto socket = weak_socket.lock();
+    AsyncExchange(*socket, Serialise(our_fob_.name(), our_fob_.Serialise()),
```

Comment:
Could we do a Jira task please, to make this possible to have data types all default serialisable to include all parts of the including name?